packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -501,14 +502,17 @@ export class CredentialsModule implements CredentialsModule {
   public findById(credentialRecordId: string): Promise<CredentialExchangeRecord | null> {
     return this.credentialRepository.findById(credentialRecordId)
   }
+
   /**
-   * Delete a credential record by id
+   * Delete a credential record by id, also calls service to delete from wallet
    *
    * @param credentialId the credential record id
+   * @param options the delete credential options for the delete operation
    */
-  public async deleteById(credentialId: string) {
+  public async deleteById(credentialId: string, options?: DeleteCredentialOptions) {
     const credentialRecord = await this.getById(credentialId)
-    return this.credentialRepository.delete(credentialRecord)
+    const service = this.getService(credentialRecord.protocolVersion)
+    return service.deleteById(credentialId, options)
```

Comment:
This is a nit, but we're retrieving the record two times now just to delete it (which wasn't the case before this PR). Could we make this a `delete` and pass the record instead of the id?